abracadabra/src/g4-mandatory/sensor_hit.hh

```diff
@@ -0,0 +1,35 @@
+#ifndef HIT_H
+#define HIT_H
+
+#include <G4VHit.hh>
+#include <G4THitsCollection.hh>
+#include <G4Allocator.hh>
+#include <G4ThreeVector.hh>
+
+
+class sensor_hit: public G4VHit {
+public:
+    sensor_hit(const G4ThreeVector& position);
+    sensor_hit(const sensor_hit&);
+    ~sensor_hit() {}
+
+    const sensor_hit& operator=(const sensor_hit&);
+    G4int operator==(const sensor_hit& other) const { return (this==&other) ? 1 : 0; }
```

Comment:
Again, puzzling and deviant code in G4: Is the author aware of something deep that I have missed, or is the author incompetent?

Shall we be daring, and write what *normal* people would write in this case?

``` c++
bool operator== (const sensor_hit& other) const { return this == &other; }
```